src/kOS.Safe/Compilation/Opcode.cs

```diff
@@ -742,12 +752,19 @@ public override void PopulateFromMLFields(List<object> fields)
             // Expect fields in the same order as the [MLField] properties of this class:
             if (fields == null || fields.Count<1)
                 throw new Exception("Saved field in ML file for BranchOpcode seems to be missing.  Version mismatch?");
-            DestinationLabel = (string)(fields[0]);  // should throw exception if not an integer-ish type.
+            // This class does something strange - it expects the KSM file to encode EITHER a string label OR an integer,
+            // but never both.  Therefore it has to determine the type of the arg to decide which it was:
+            if (fields[0] is string)
+                DestinationLabel = (string)(fields[0]);
+            else
+                Distance = (int)fields[0];
         }
 
         public override string ToString()
         {
-            return string.Format("{0} {1}", Name, Distance);
+            // Format string forces printing of '+/-' sign always, even for positive numbers.
+            // The intent is to make it more clear that this is a relative, not absolute jump:
+            return string.Format("{0} {1:+#;-#;+0}", Name, Distance);
```

Comment:
This change was not strictly needed for the fix, but as long as my fingers were in this code, I changed it because I think this communicates the "relativeness" of the jump better to the user.